react-components/src/components/RuleBasedOutputs/RuleBasedOutputsPanel.tsx

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import styled from 'styled-components';
+
+import { Flex } from '@cognite/cogs.js';
+import { type RuleOutput } from './types';
+import { useState, type ReactElement } from 'react';
+import { withSuppressRevealEvents } from '../../higher-order-components/withSuppressRevealEvents';
+import { ShowMore } from './components/OutputPanelShowMore';
+import { OutputPanelItem } from './components/OutputPanelItem';
+
+const MAX_WHEN_NO_SHOW_MORE = 5;
+const MAX_HEIGHT = '300px';
+
+type RuleBasedOutputsPanelProps = {
+  ruleOutputs?: RuleOutput[];
+};
+
+export const RuleBasedOutputsPanel = ({
+  ruleOutputs
+}: RuleBasedOutputsPanelProps): ReactElement => {
+  const [isShowMore, setIsShowMore] = useState(false);
+
+  return (
+    <>
+      {ruleOutputs !== undefined && ruleOutputs?.length > 0 && (
+        <StyledRuleBasedOutputsPanel $showMore={isShowMore}>
+          <Flex direction="column" gap={14}>
+            {ruleOutputs.map((output, index) => {
+              return (
+                <>
+                  {isShowMore || index <= MAX_WHEN_NO_SHOW_MORE - 1 ? (
+                    <OutputPanelItem index={index} output={output} />
+                  ) : (
+                    <></>
+                  )}
+                </>
+              );
+            })}
+            {ruleOutputs.length > MAX_WHEN_NO_SHOW_MORE && (
+              <ShowMore
+                isShowMore={isShowMore}
+                onShowMore={() => {
+                  setIsShowMore((previousState) => !previousState);
+                }}
+              />
+            )}
```

Comment:
Have you had the chance to test the "show more"-functionality? Is there any project with a ruleset containing that many rules?

Reply:
I tested by creating 8 rule outputs in some ruleSets as the max is 5 atm